src/modules/navigator/rtl.cpp

```diff
@@ -555,21 +573,17 @@ void RTL::advance_rtl()
 	const bool vtol_in_fw_mode = _navigator->get_vstatus()->is_vtol
 				     && _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING;
 
+	// determines if the vehicle should descent in MR before land
+	const bool mr_descent = _param_rtl_descend_mr.get() > ALT_SIGMA;
+
 	switch (_rtl_state) {
 	case RTL_STATE_CLIMB:
 		_rtl_state = RTL_STATE_RETURN;
 		break;
 
 	case RTL_STATE_RETURN:
 		setClimbAndReturnDone(true);
-
-		if (vtol_in_fw_mode || descend_and_loiter) {
-			_rtl_state = RTL_STATE_DESCEND;
-
-		} else {
-			_rtl_state = RTL_STATE_LAND;
-		}
-
+		_rtl_state = RTL_STATE_DESCEND;
```

Comment:
By not checking the `descend_and_loiter` condition here, we are deviating slightly from the upstream behavior for the `RTL_DESCEND` parameters. This can be confusing since it is very close to upstream but not exactly the same.

Have you considered swapping the existing logic for two parameters that only work in their respective VTOL modes?: e.g. `RTL_FW_DESCEND_ALT` and `RTL_MR_DESCEND_ALT`. This would be a bigger change, but with more predictable behavior IMO.

Reply:
I will try to get the changes upstream as well if we decide to use them. In my opinion, the current behavior is wrong, or at least confusing considering the parameter descriptions. I see no reason for the MR RTL pattern to be fundamentally different for a loiter time of 0 versus 1 second, for example.

I have considered something like RTL_FW_DESCEND_ALT/RTL_MR_DESCEND_ALT, but there is some issues, like if we need separate loitering time for the two, which may make sense. Anyway, you probably want to be able to use both in FW. First FW loiter down to X meters, then fast a MR descend to Y meters, before a slow landing.

I can have a quick look at how it would be.

Reply:
I agree that pushing the changes upstream is a good idea, and that counters the point of differing behavior. My main issue with the way this works now is that a multirotor drone will move `RTL_ALT -> RTL_DESCEND_ALT -> RTL_DESCEND_MR -> LAND` which feels unintuitive to me.

Reply:
Would you prefer like this:
FW RTL: `RTL_ALT -> RTL_FW_DESCEND_ALT -> TRANSITION_MC -> RTL_MC_DESCEND_ALT -> LAND`
MC RTL: `RTL_ALT -> RTL_MC_DESCEND_ALT -> LAND`

We could do that, but need to decide on loiter as well then. Should there be one or two loiter points?

Reply:
A bit simplified state diagrams.

Current aviant/1.12:
![rtl](https://user-images.githubusercontent.com/108630475/214285219-1535fd7a-a1b7-4ad4-aba1-8f870891cbbc.png)

This PR:
![rtl2](https://user-images.githubusercontent.com/108630475/214285238-db849aa3-c370-459d-b37a-7da6260a4081.png)

Reply:
For FW loitering only, and if in MR skip RTL_DESCEND_ALT:
![rtl3](https://user-images.githubusercontent.com/108630475/214324074-6e97c53c-faa2-47e8-bcad-b6affa35f25e.png)

Reply:
The 3rd state machine is now implemented.